runtime/executor/tensor_parser_exec_aten.cpp

```diff
@@ -111,12 +111,62 @@ ET_NODISCARD Result<BoxedEvalueList<executorch::aten::Tensor>> parseTensorList(
       evalp_list, tensor_list, tensor_indices->size());
 }
 
+ET_NODISCARD Error validateExternalTensor(
```

Comment:
Nothing about this is specific to external tensors per se; `validateTensorLayout` (and renaming the param to something generic like `expected_layout`) would make that more clear.